### DISCUSSION TITLE
Announcement: JDBCParquetWriter for writing ResultSets to Parquet Files

### DISCUSSION THREAD

**Discussion by manticore-projects** [ClickHouse/ClickHouse#48819] [Show and tell]
Greetings.

Since I want to test/use ClickHouse and did not find anything good for the migration from Oracle and friends, I wrote a small library:
https://manticore-projects.com/JDBCParquetWriter/index.html
https://github.com/manticore-projects/JDBCParquetWriter

Its a first working version, but it seems to get the Type/Column translation right. Also it uses latest Hadoop/Parquet and supersedes a lot of deprecated samples found in the internet.

Feedback will be most appreciated.

I plan to write a "Schema to ClickHouse" method, which exports all tables of a schema into parquet files and provides the ClickHouse specific DDL and `INSERT INTO ... SELECT * FROM read_parquet()` commands (including indexes).

**Comment by l1t1** [comment]
nice job
